woocommerce/woocommerce-ios
# Products Onboarding: Add the banner experiment to the app

Closes: #8002

## Description

This PR adds the A/B test for the products onboarding banner, and removes it from behind the feature flag.

The banner will only be visible to eligible users in the A/B test treatment group, and only after the A/B test is started on the backend. (The banner is also visible to eligible a8c users in the treatment group while the test is in staging.)

## Changes

* Adds the products onboarding banner experiment.
* Replaces the feature flag check with a check for the A/B test treatment group.
* Adds `MockABTesting` for unit testing A/B test behavior:
   * `setVariation(_:for:)` lets us mock an experiment assignment in unit tests by manually setting the provided variation (control or treatment) for the given experiment.
   * This new method is used in our unit tests so we can still check that the products onboarding banner takes precedence when the user is eligible to see it.
* Removes the unit test that was used for checking the banner behavior behind a feature flag.

## Testing

**Prerequisite (before launching the app)**

1. Open the experiment details in Abacus (linked at the top of pbxNRc-26F-p2 or ping me for the direct link).
2. In the Audience section of the experiment setup, go to Variations > Manual Assignment and use the bookmarklet to assign yourself to the **treatment** group.

**Test the treatment behavior**

1. Build and run the app. **Note:** You must be logged in to the app with an a8c account to test the experiment assignment/behavior while the experiment is in staging. If you are logged in to a different account, log out and log in again with your a8c account.
2. Select a store with no products.
3. Confirm you see the products onboarding banner on the My Store dashboard.

**Test the control behavior**

1. Build and run the app.
2. Log out and log in with a non-a8c test account. (Alternately, use the prerequisite steps to assign yourself to the **control** group to use your a8c account. While the test is in staging, manually assigning yourself to the control group or logging in with a non-a8c account will show you the control behavior.)
3. Select a store with no products.
4. Confirm you don't see any banner on the My Store dashboard.

## Submitter Checklist

Update release notes:
